syntax = "proto3";

message GreetingRequest {
    int32 user_id = 1;
}

message GreetingResponse {
    string greetings = 1;
}

message AuthorizeRequest {
}

message AuthorizeResponse {
    repeated string roles = 1;
    bool is_superuser = 2;
}

service Auth {
    rpc Greet(GreetingRequest) returns (GreetingResponse);
    rpc Authorize(AuthorizeRequest) returns (AuthorizeResponse);
}